### ISSUE TITLE
Investigate interpreter segfault TODO comment for Python 3.6 vs 3.11

### ISSUE THREAD

**Issue by korydraughn** [irods/python-irodsclient#553]
- [x] Stub out `tearDownClass` (if versions >3.8 don't have the problem)

---

See comment in the following code.
https://github.com/irods/python-irodsclient/blob/86a8f11a9399db29774c4096d83bba733a024ab6/irods/test/data_obj_test.py#L741-L744

This may apply to other tests (i.e. collection_test.py).

**Comment by alanking** [comment]
FWIW, I would recommend dropping support for 3.6 as part of #480 as it has been EOL since the end of 2021, and so we won't need to worry about this.

**Comment by trel** [comment]
Hmm, that matrix of dropping support might be confusing to users.

**Comment by trel** [comment]
So perhaps we move this issue to 3.0.0 and take care of all EOL issues at once?

**Comment by alanking** [comment]
I was more thinking of dropping support in 3.0.0. If we drop support for anything that came *after* python 2.7, that would be kind of weird.

Yes, I think moving this to the 3.0.0 milestone makes sense.

**Comment by trel** [comment]
3.0.0 - done.

**Comment by trel** [comment]
We want to remove support for  `< Python 3.8` for PRC 3.0.0.

Aka, we'll support 3.8 and later.

**Comment by d-w-moore** [comment]
> We want to remove support for `< Python 3.8` for PRC 3.0.0.
> 
> Aka, we'll support 3.8 and later.

Does this still apply? in other words,, though in the current version of the issue description @alanking has given Python 3.9 as the cutoff for our minimally supporting, I kind of think we should still support  3.8+ in PRC as 3.8  is the system-Python on ubuntu 20.04 and we still actively publish packages for that OS.

**Comment by d-w-moore** [comment]
Update, Python 3.8.0 still segfaults. Going on to 3.9, since ubuntu20.04 has ready-installable packages for that version.

**Comment by d-w-moore** [comment]
Ok, I've established 3.9.5 (the version of system Python 3.9 that is ready installable on ubuntu 20.04) does not segfault for this case.

**Comment by d-w-moore** [comment]
Ok, in the clear for 3.9.0.  We'll work on checking off the boxes and make Python 3.9 the cutoff for upcoming PRC v3.1.0

**Comment by trel** [comment]
> make Python 3.9 the cutoff for upcoming PRC v3.1.0

Yes please.

The packages/OSs we publish for the server are independent from this client library.

**Comment by d-w-moore** [comment]
> > make Python 3.9 the cutoff for upcoming PRC v3.1.0
> 
> Yes please.
> 
> The packages/OSs we publish for the server are independent from this client library.

True, although there will be a number of new things for e.g. Ubuntu 20 system admins to watch now like making sure an SSL development support lib is installed (think there are a couple options for that one?) as well as zlib1g-dev and maybe others before building a custom Python. Otherwise pip won't install.

**Comment by trel** [comment]
But i'm pretty sure those are way outside our scope for this library, right?

If you think not, then let's open a separate issue to discuss any particular pitfalls.

**Comment by d-w-moore** [comment]
On re-examination, yes - our README is presently just a Python iRODS Client usage guide , not including directions to management of the OS's facilities (like builds of Python) and packages that we depend on.  It's big enough already, and if there's a need later for a `README-install.md` we'll know it.

**Comment by alanking** [comment]
@d-w-moore confirmed that Python versions >3.8 do not exhibit the issue here. The next release will be listed as compatible only with Python versions >3.8, so the `tearDownClass` method has been stubbed out and the TODO comment has been removed.

If that sounds accurate, please close this one, @d-w-moore. Thanks

**Comment by trel** [comment]
looks good.  closing.
